.github/workflows/build.yml

```diff
@@ -20,33 +24,35 @@ jobs:
       - uses: gradle/wrapper-validation-action@v1
       - uses: gradle/gradle-build-action@v2
         with:
-          arguments: |
-            build
-            projectHealth
-            mergeLintSarif
-            mergeDetektSarif
-            :plugins:buildPlugins
-            --continue
-      # checks integrity of release tasks, update publish workflows too if does not work
-      - uses: gradle/gradle-build-action@v2
-        with:
-          arguments: |
-            publishAppyxReleasePublicationToOSSRHRepository
-            publishAppyxReleasePublicationToSonatypeSnapshotRepository
-            --dry-run
-            --no-parallel
-      - uses: gradle/gradle-build-action@v2
-        if: github.ref == 'refs/heads/main' && github.repository == 'bumble-tech/appyx'
+          cache-read-only: ${{ env.MAIN_BRANCH != 'true' }}
```

Comment:
Are we okay with having two different `main` branches updating the cache? surely they will conflict with one another?

Reply:
I expect the cache to be merged.

Reply:
Is that ideal? what if the 2.x branch changes dependencies ahead of the 1.x branch? They could constantly override each other

Reply:
Tried to switch a version of Coil multiple times and to recompile sandbox, got cache hits for both Coil versions.

Maybe approach is not ideal, but I can't provide anything better, `gradle-build-action` does not allows to define additional keys.

Reply:
fair enough